src/formatDate.js

```diff
@@ -50,7 +50,59 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let FORMAT_YEAR = '';
+  let FORMAT_MONTH = '';
+  let FORMAT_DAY = '';
+  const dateFormatNew = [];
+
+  const splitOldDate = fromFormat[fromFormat.length - 1];
+  const splitNewDate = toFormat[toFormat.length - 1];
+
+  const dateFormatOld = date.split(splitOldDate);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const elementArray = fromFormat[i];
+
+    if (elementArray === 'YYYY') {
+      FORMAT_YEAR = dateFormatOld[i];
+    } else if (elementArray === 'YY') {
+      FORMAT_YEAR = dateFormatOld[i];
+    } else if (elementArray === 'MM') {
+      FORMAT_MONTH = dateFormatOld[i];
+    } else if (elementArray === 'DD') {
+      FORMAT_DAY = dateFormatOld[i];
+    }
```

Comment:
```suggestion
    const part = fromFormat[i];
    
    switch (part) {
      case 'YY':
      case 'YYYY':
        year = dateValuesOld[i];
        break;
      case 'MM':
        month = dateValuesOld[i];
        break;
      case 'DD':
        day =  dateValuesOld[i];
        break;
    }
```
Consider using `switch` as a better option.